File: src/client-app/src/models/requests/transactions/addOutcomeTransactionCommand.ts

export interface AddOutcomeTransactionCommand {
    sourceAccountId: string
    recipientName: string
    transactionAmount: number
    categoryValue: string
    subcategoryValue: string
    transactionDateTime: Date
}

export class AddOutcomeTransactionCommand implements AddOutcomeTransactionCommand {
    constructor(values?: AddOutcomeTransactionCommand) {
        this.sourceAccountId = values?.sourceAccountId || ''
        this.recipientName = values?.recipientName || ''
        this.transactionAmount = values?.transactionAmount || 0
        this.categoryValue = values?.categoryValue || ''
        this.subcategoryValue = values?.subcategoryValue || ''
        this.transactionDateTime = values?.transactionDateTime || new Date()
    }
}

File: src/PennyPlanner.Application/Transactions/AddOutcomeTransaction/AddOutcomeTransactionCommand.cs
using CommonAbstractions.DB.Messaging;
using PennyPlanner.Domain.Transactions;

namespace PennyPlanner.Application.Transactions.AddOutcomeTransaction
{
    public sealed record AddOutcomeTransactionCommand(
        string SourceAccountId,
        string RecipientName,
        decimal TransactionAmount,
        string CategoryValue,
        string SubcategoryValue,
        DateTime TransactionDateTime) : ICommand<Transaction>
    {
    }
}
